plugins/modules/system/beadm.py

```diff
@@ -165,7 +165,7 @@ def _find_be_by_name(self, out):
         if '@' in self.name:
             for line in out.splitlines():
                 if self.is_freebsd:
-                    check = re.match(r'.+/({0})\s+\-'.format(self.name), line)
+                    check = re.match(r'.+/({0})\s+\-'.format(re.escape(self.name)), line)
                     if check:
                         return check
```

Comment:
I'm not sure if returning `check` here (a RE match result) is correct. If the caller does `result[1]` like in line 201, it will return `self.name`. But if the caller does `result[2]` like in line 257, it will result in a `IndexError: no such group` exception raised.

Reply:
Indeed, line 257 will raise the exception.
Unfortunately, I have no idea how to fix it as I don't understand line 168 (regex) at all. Probably, this regex should be changed to include the second group, needed by code in line 257.
I contacted the author of the module and the committer of [https://github.com/ansible/ansible/commit/e1d0795aa3b234365c2bd7c0c9c3bce36a4964e2](https://github.com/ansible/ansible/commit/e1d0795aa3b234365c2bd7c0c9c3bce36a4964e2) 
Hope they can help.